pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -36,6 +36,8 @@ const (
 func parseConfig(config *zconfig.EdgeDevConfig, getconfigCtx *getconfigContext,
 	usingSaved bool) bool {
 
+	parseCipherContextConfig(getconfigCtx, config)
+
```

Comment:
I still think this should be down on Line 59 i.e. just before parseDataStoreConfig.

But as I've said, this will not prevent the network (or datastore) config from racing past the ciphercontext because that depends on the behavior of the select loop in nim and downloader, respectively.
So we can do a short-term fix with this on line 59, but we still need code in nim, downloader and domainmgr (latter for encrypted user data) to handle the ciphercontext arriving late.
So can yo create a story for that follow-on work?

Reply:
Currently, I am testing out a fix. I will also test with your suggested changes and will see is there any difference or not. My assumption is that we need to parse the new cipher contexts before returning the reboot flag true. I will test both the changes. I will create stories for adding the handler in nim, downloader and domainmgr.

Reply:
Why do you need new ciphers if you are in the process of rebooting? In that case we should minimize changes to the config.

Reply:
Why do you need new ciphers if you are in the process of rebooting? In that case we should minimize changes to the config.

Reply:
Yeah, it makes sense. I updated the PR accordingly.